File: Backend/app/api/v1/routes/history.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.services.history_service import HistoryService

router = APIRouter()


def get_history_service(db: Session = Depends(get_db)) -> HistoryService:
    return HistoryService(db)


@router.get("")
def list_history(keyword: str | None = None, service: HistoryService = Depends(get_history_service)) -> dict[str, list[dict[str, object]]]:
    return {"items": service.list_history(keyword=keyword)}


@router.get("/{history_id}")
def get_history(history_id: str, service: HistoryService = Depends(get_history_service)) -> dict[str, object]:
    for item in service.list_history():
        if str(item.get("project_id")) == history_id or str(item.get("title")) == history_id:
            return item
    return {"id": history_id, "status": "not_found"}

File: frontend/src/services/history.ts
import { api } from './api'

export interface HistoryRecord {
  id?: string
  time: string | null
  project_id: string
  project: string
  step: string
  summary: string
  title: string
  content_text?: string
}

export interface HistoryDetail extends HistoryRecord {
  content_text: string
}

export async function listHistory(params: { keyword?: string } = {}) {
  const { data } = await api.get<{ items: HistoryRecord[] }>('/history', { params })
  return data.items
}

export async function getHistory(historyId: string) {
  const { data } = await api.get<HistoryDetail | { id: string; status: string }>(`/history/${historyId}`)
  return data
}
